nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/blob/AESBlobStore.java

```diff
@@ -0,0 +1,308 @@
+/*
+ * (C) Copyright 2006-2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Florent Guillaume
+ */
+package org.nuxeo.ecm.core.blob;
+
+import static java.nio.charset.StandardCharsets.US_ASCII;
+import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;
+
+import java.io.BufferedOutputStream;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.FilterInputStream;
+import java.io.FilterOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.security.GeneralSecurityException;
+import java.security.Key;
+import java.security.SecureRandom;
+import java.util.Arrays;
+import java.util.Random;
+
+import javax.crypto.Cipher;
+import javax.crypto.CipherInputStream;
+import javax.crypto.CipherOutputStream;
+
+import org.apache.commons.io.IOUtils;
+
+/**
+ * A blob store that encrypts binaries on the filesystem using AES.
+ *
+ * @since 11.1
+ */
+public class AESBlobStore extends LocalBlobStore {
+
+    protected static final byte[] FILE_MAGIC = "NUXEOCRYPT".getBytes(US_ASCII);
+
+    protected static final int FILE_VERSION_1 = 1;
+
+    protected static final int USE_KEYSTORE = 1;
+
+    protected static final int USE_PBKDF2 = 2;
+
+    /**
+     * If {@code true}, use the insecure AES/CBC/PKCS5Padding for encryption. The default is {@code false}, to use
+     * AES/GCM/NoPadding.
+     */
+    protected static final String PARAM_KEY_USE_INSECURE_CIPHER = "useInsecureCipher";
```

Comment:
i think this a duplicate of `org.nuxeo.ecm.core.blob.AESBlobStoreConfiguration#PROP_KEY_USE_INSECURE_CIPHER` [here](https://github.com/nuxeo/nuxeo/blob/feature-NXP-27435-record-retention-hold-10.10/nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/blob/AESBlobStoreConfiguration.java#L89)

Reply:
👍